packages/iota/src/validation/nativeTokens.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+import bigInt from "big-integer";
+import { MAX_NATIVE_TOKEN_COUNT } from "../binary/nativeTokens";
+import type { INativeToken } from "../models/INativeToken";
+import { ValidationHelper } from "../utils/validationHelper";
+import { IValidationResult, failValidation, mergeValidationResults } from "./result";
+
+/**
+ * Validate native tokens.
+ * @param object The object to validate.
+ * @returns The validation result.
+ */
+export function validateNativeTokens(object: INativeToken[] | undefined): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (object) {
+        const tokenIds = object.map(token => {
+            result = mergeValidationResults(result, validateNativeToken(token));
```

Comment:
Split this side work from the tokenId mapping, and do it at the bottom of the method.